### DISCUSSION TITLE
Thoughts on time-travel?

### DISCUSSION THREAD

**Discussion by tluyben** [rabbibotton/clog#359] [Ideas]
In the sense of state, this is a lovely thing in for instance ELM https://elm-lang.org/news/time-travel-made-easy (it's possible to more or less extent in other technologies). However as CLOG is also for frontend but the entire state (backend & frontend) are on the server, it seems it be an interesting. I guess it would work if one flags which parts are state.

**Comment by rabbibotton** [comment]
I have been working exploring snapshotting for clog and software testing for graphical clog apps both are direct requirements for time travel debugging to work for CL and CLOG. This does impress me to push the time table on making something like this possible! Thanks!
